File: August 2022/August_18.cpp
class Solution {
public:
    int minSetSize(vector<int>& arr) {
        unordered_map<int, int> mp;
        for(int i : arr)
            mp[i]++;
        priority_queue<pair<int, int> > pq;
        for(auto p : mp)
            pq.push({p.second, p.first});
        int rem = 0, ans = 0;
        while(rem < arr.size() / 2){
            pair p = pq.top();
            pq.pop();
            rem += p.first;
            ans++;
        }
        return ans;
    }
};

File: August 2022/August_18.java
class Solution {
    public int minSetSize(int[] arr) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int i : arr)
            map.put(i, map.getOrDefault(i, 0) + 1);
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> b[0] - a[0]);
        for(int i : map.keySet())
            pq.add(new int[]{map.get(i), i});
        int rem = 0, ans = 0;
        while(rem < arr.length / 2){
            rem += pq.remove()[0];
            ans++;
        }
        return ans;
    }
}
